server.go

```diff
@@ -537,7 +556,10 @@ func compilePath(path spec.Path) (*regexp.Regexp, []string) {
 		if submatches == nil {
 			pattern += `/` + part
 		} else {
-			pattern += `/(?P<` + submatches[0][1] + `>[\w-_.]+)`
+			// Special characters as defined by:
+			//
+			// https://tools.ietf.org/html/rfc3986#section-3.3
+			pattern += `/(?P<` + submatches[0][1] + `>[\w%-._~!$&'()*+,;=]+)`
```

Comment:
I suppose we are looking at the `segment-nz-nc`, so should we also allow `@` here?
```
segment-nz-nc = 1*( unreserved / pct-encoded / sub-delims / "@" )
                    ; non-zero-length segment without any colon ":"
```

Reply:
Also tested that `@` is allowed in our API.
```
curl https://api.stripe.com/v1/plans/Pro+plan+%24699%2Fmo@nth \
  -u sk_test_4eC39HqLyjWDarjtT1zdp7dc:
```

Reply:
Oh shoot, you're right! And `:` is allowed too. Just added those in.